File: pingoygou-parent/pinyougou-portal-web/src/main/resources/resources/js/controller/contentController.js
app.controller("contentController",function ($scope,$controller,contentService) {

    //控制器的继承
    $controller('publicController',{$scope:$scope});

    //轮播广告的存放地方
    $scope.contentList=[]
    $scope.findByCategoryId=function (categoryId) {
        contentService.findByCategoryId(categoryId).success(function (response) {
            $scope.contentList[categoryId]=response;
        });
    }
});

File: pingoygou-parent/pinyougou-portal-web/src/main/java/com/pinyougou/protal/controller/ContentController.java
package com.pinyougou.protal.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.content.service.ContentService;
import com.pinyougou.pojo.TbContent;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/content")
public class ContentController {

    @Reference
    private ContentService contentService;

    @RequestMapping("findByCategoryId.do")
    public List<TbContent> findByCategoryId(Long categoryId){
        List<TbContent> lists = contentService.findByCategoryId(categoryId);
        return lists;
    }
}
